### PR TITLE
R4R: Fix service

### PR DESCRIPTION
**Pull request by secret2830**
Resolves: #2285

### CODE REVIEW

File: app/v3/service/internal/keeper/invocation.go [hunk 1]
```diff
@@ -246,7 +246,9 @@ func (k Keeper) StartRequestContext(
 	requestContext.State = types.RUNNING
 	k.SetRequestContext(ctx, requestContextID, requestContext)
 
-	if requestContext.BatchState == types.BATCHCOMPLETED {
+	if requestContext.BatchState == types.BATCHCOMPLETED &&
```
Comment (chengwenxi) [inline app/v3/service/internal/keeper/invocation.go]: if the `NewRequestBatch` is the next batch, it seems better to remove it when pausing request context.

Comment (secret2830) [reply app/v3/service/internal/keeper/invocation.go]: My consideration: 
1. Removing needs another key
2. Low performance if repeated paused and started before timeout

Naturally deleted instead

Comment (chengwenxi) [reply app/v3/service/internal/keeper/invocation.go]: Add more comments for this code.
It's confused about
```golang
requestContext.BatchRequestCount == 0 || requestContext.BatchRequestCount != requestContext.BatchResponseCount
```

Comment (chengwenxi) [reply app/v3/service/internal/keeper/invocation.go]: or add a new batch state to describe this case.

Comment (chengwenxi) [reply app/v3/service/internal/keeper/invocation.go]: todo in https://github.com/irisnet/irishub/pull/2293

**Pull request merged**